### PR TITLE
Python configargparse

### PR DESCRIPTION
**Pull request by egpbos**
- [x] I followed the [CONTRIBUTING guidelines](../blob/master/CONTRIBUTING.md).

Below, describe what this Pull Request adds:

Added an I/O section with an item for configuration using either argparse or ConfigArgParse. Using the latter in the Spoken Language project now, really happy with it!

### CODE REVIEW

**Comment by egpbos** [comment]
Oops, I also included some commits from another branch. I was doing a general update sweep of the Python chapter there. Maybe fine to include those here as well though, since I've been sitting on them for 3 months already :P

File: best_practices/language_guides/python.md [hunk 1]
```diff
@@ -280,3 +296,7 @@ We recommend `flask`.
 * [nltk](http://www.nltk.org/) Natural Language Toolkit
 * [Pattern](http://www.clips.ua.ac.be/pattern): web/text mining module
 * [gensim](https://radimrehurek.com/gensim/): Topic modeling
+
+### I/O
```
Comment (bouweandela) [inline best_practices/language_guides/python.md]:
```suggestion
### Creating programs with command line options
```
would be a bit easier to understand for most people

File: best_practices/language_guides/python.md [hunk 2]
```diff
@@ -280,3 +296,7 @@ We recommend `flask`.
...
+
+* For run-time configuration via command-line options, the built-in `argparse` module usually suffices. A more complete solution is [`ConfigArgParse`](https://github.com/bw2/ConfigArgParse). This (almost) drop-in replacement for `argparse` allows you to not only specify configuration options via command-line options, but also via (ini or yaml) configuration files and via environment variables.
```
Comment (bouweandela) [inline best_practices/language_guides/python.md]:
```suggestion
* For run-time configuration via command-line options, the built-in [`argparse`](https://docs.python.org/library/argparse.html) module usually suffices.
* A more complete solution is [`ConfigArgParse`](https://github.com/bw2/ConfigArgParse). This (almost) drop-in replacement for `argparse` allows you to not only specify configuration options via command-line options, but also via (ini or yaml) configuration files and via environment variables.
* Other popular libraries are [`click`](https://google.github.io/python-fire/) and [`fire`](https://google.github.io/python-fire/).

```

File: best_practices/language_guides/python.md (hunk 1 above)
Comment (egpbos) [reply best_practices/language_guides/python.md]: you're right, i was anticipating a broader paragraph with other I/O modules like things for hdf5, but this is better

File: best_practices/language_guides/python.md (hunk 2 above)
Comment (egpbos) [reply best_practices/language_guides/python.md]: Yep, much better!

**Pull request merged**